eclipse-theia/theia
# 7989-refactor-preview: Support for Refactor Preview functionality

Signed-off-by: Dan Arad <dan.arad@sap.com>

#### What it does
Resolves #7989 - Adds Refactor Preview functionality.

Remarks:
These features existing in vscode were not implemented currently in this PR:
1. Checkboxes for selecting which changes to implement
2. Upon right clicking on text line in Refactor Preview pane - context menu is not displayed like in vscode.

CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=23016

#### How to test
1. Open ts file, click F2 for renaming a function
2. Enter new name and press Shift+Enter
3. Refactor Preview pane should open in bottom with changes
4. Press V button on pane toolbar to accept and apply changes
5. Press X button to discard changes

#### Review checklist


#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
